types/mongodb/index.d.ts

```diff
@@ -240,57 +241,120 @@ export interface MongoClientOptions extends
     SocketOptions,
     SSLOptions,
     HighAvailabilityOptions {
-    // The logging level (error/warn/info/debug)
+
+    /**
+     * The logging level (error/warn/info/debug)
+     */
     loggerLevel?: string;
-    // Custom logger object
-    logger?: Object;
-    // Default: false;
+
+    /**
+     * Custom logger object
+     */
+    logger?: Object | log;
+
+    /**
+     * Validate MongoClient passed in options for correctness.
+     * Default: false
+     */
     validateOptions?: Object;
-    // The name of the application that created this MongoClient instance.
-    appname?: string;
+
+    /**
+     * The name of the application that created this MongoClient instance.
+     */
+     appname?: string;
+
+     /**
+      * Authentifikation credentials
+      */
     auth?: {
+        /**
+         * The username for auth
+         */
         user: string;
+        /**
+         * The password for auth
+         */
         password: string;
-    },
-    // Determines whether or not to use the new url parser
-    useNewUrlParser?: boolean
+    };
+
+    /**
+     * Determines whether or not to use the new url parser
+     */
+     useNewUrlParser?: boolean;
+
+    /**
+     * Mechanism for authentication: MDEFAULT, GSSAPI, PLAIN, MONGODB-X509, or SCRAM-SHA-1
+     */
+    authMechanism?: 'MDEFAULT' | 'GSSAPI' | 'PLAIN' | 'MONGODB-X509' | 'SCRAM-SHA-1' | string;
```

Comment:
Valid values for `authMechanism` are  `DEFAULT` (Note there is no M), `GSSAPI`, `PLAIN`, `MONGODB-CR`, `MONGODB-X509`, `SCRAM-SHA-1`, and `SCRAM-SHA-256`. See https://github.com/mongodb/node-mongodb-native/blob/master/lib/authenticate.js#L34

Reply:
Ok, you're right, I fixed it.